graylog2-web-interface/src/components/common/OverlayDropdown.tsx

```diff
@@ -50,9 +51,23 @@ const FilterProps = ({ children, style }: _FilterProps) => (
   </>
 );
 
-const OverlayDropdown = ({ children, menuContainer, onToggle, placement, show, toggle }) => {
-  const [currentPlacement, setCurrentPlacement] = useState(placement);
-  const toggleTarget = React.createRef<HTMLSpanElement>();
+FilterProps.defaultProps = {
+  style: {},
+};
+
+type Props = {
+  children: React.ReactNode,
+  menuContainer?: HTMLElement,
+  onToggle: () => void,
+  placement?: Placement,
+  show: boolean,
+  toggleChild?: React.ReactNode,
+  renderToggle?: (payload: { onToggle: () => void, toggleTarget: React.Ref<HTMLElement> }) => React.ReactNode,
+}
```

Comment:
I tried to type the props like this
```
type BaseProps = {
  children: React.ReactNode,
  menuContainer?: HTMLElement,
  onToggle: () => void,
  placement?: Placement,
  show: boolean,
}

type CustomToggleProps = BaseProps & {
  hasCustomToggleWrapper: true,
  renderToggle: (payload: { onToggle: () => void, toggleTarget: React.Ref<HTMLElement> }) => React.ReactNode,
}

type DefaultToggleProps = BaseProps & {
  hasCustomToggleWrapper: false,
  toggleChild: React.ReactNode,
}

type Props = CustomToggleProps | DefaultToggleProps
```

But could not get rid of errors like:
```
TS2339: Property 'toggleChild' does not exist on type '{ renderToggle: (payload: { onToggle: () => void; toggleTarget: Ref<HTMLElement>; }) => ReactNode; } | { toggleChild: ReactNode; }'.   Property 'toggleChild' does not exist on type '{ renderToggle: (payload: { onToggle: () => void; toggleTarget: Ref<HTMLElement>; }) => ReactNode; }'.
```

Even when wrapping the related code in `!!hasCustomToggleWrapper`